lib/support/storageManager.js

```diff
@@ -6,16 +6,29 @@ const fs = require('fs-extra'),
   moment = require('moment'),
   path = require('path'),
   url = require('url'),
+  zlib = require('zlib'),
   crypto = require('crypto'),
   urlParser = require('url');
 
 Promise.promisifyAll(fs);
+Promise.promisifyAll(zlib);
+
 const mkdirp = Promise.promisify(require('mkdirp'));
 
-function write(dirPath, filename, data) {
+function write(dirPath, filename, data, gzip) {
   return Promise.join(dirPath, filename, data,
-    (dirPath, filename, data) =>
-      fs.writeFileAsync(path.join(dirPath, filename), data, 'utf8'));
+    (dirPath, filename, data) => {
+      if (gzip) {
+        const buff = new Buffer(data, 'utf8');
+        return zlib.gzipAsync(buff, {level: 1}).then((buffer) =>
+          fs.writeFileAsync(path.join(dirPath, filename + '.gz'), buffer, 'utf8')
+        );
+      }
```

Comment:
Just a nitpick, but this isn't how we typically place the braces for else, is it?